### PR TITLE
examples:  update tensorflow backdoor poisoning notebooks to new deployment method

### PR DESCRIPTION
**Pull request by alexb1200**
Towards closing #152 

- Added new deployment instructions and documentation to README
- Moved custom plugins to  dioptra/examples/task-plugins/custom-poisoning_plugins
- Moved MLproject to src folder
- Modified demo-mnist-poison-backdoor-adv-embedding, demo-mnist-poison-backdoor-baseline, demo-mnist-poison-backdoor-clean-label to use the new scripts. 
- Removed unused files in example folder

### CODE REVIEW

**Review by jtsextonMITRE — COMMENTED** [review]
`src/train.py`, 
`src/train_with_madry_pgd_defense.py`, 
`src/spatial_smoothing.py`, 
`src/jpeg_compression.py`, 
`src/gen_poison_model.py`, 
`src/gaussian_augmentation.py` 
 
should have the `load_dataset_from_mlruns` functionality extracted to a task plugin or just added to into the workflow if all that functionality already exists.

**Comment by jtsextonMITRE** [comment]
`response_gen_poison_le_net_train_clean` <- This code block in demo-mnist-poison-backdoor-clean-label.ipynb should have

`depends_on=response_le_net_train_robust` instead of `depends_on=response_le_net_train`

**Comment by jtsextonMITRE** [comment]
`response_infer_pos_model_clean` <- This code block has a mismatch between the depends_on (`response_le_net_train_backdoor_model_clean`) and the run_id (`response_gen_poison_le_net_test`)being used.

**Comment by jtsextonMITRE** [comment]
`response_infer_reg_model` <- This code block has a mismatch between the depends_on (`response_le_net_train`) and the run_id (`response_gen_poison_le_net_test`) being used.

**Comment by jtsextonMITRE** [comment]
The `demo-mnist-poison-backdoor-clean-label.ipynb` notebook has a title of Adversarial Embedding

**Comment by jtsextonMITRE** [comment]
We should add a comment near `response_gen_poison_le_net_train_clean` in `...clean-label.ipynb` that the poison fraction, the test dataset or the batch size being small can result in errors due to ART potentially randomly selecting 0 images to poison.

File: examples/tensorflow-backdoor-poisoning/demo-mnist-poison-backdoor-adv-embedding.ipynb
Comment (jkglasbrenner) [inline examples/tensorflow-backdoor-poisoning/demo-mnist-poison-backdoor-adv-embedding.ipynb:1]: Remove the USERNAME = "" from the code block at the beginning and fix EXPERIMENT_NAME to reflect change.

Comment (jkglasbrenner) [inline examples/tensorflow-backdoor-poisoning/demo-mnist-poison-backdoor-adv-embedding.ipynb:1]: Remove the MLFlow query section at the end of the notebook.

Comment (jkglasbrenner) [reply examples/tensorflow-backdoor-poisoning/demo-mnist-poison-backdoor-adv-embedding.ipynb:1]: This can be applied to all 3 notebooks.

Comment (jkglasbrenner) [reply examples/tensorflow-backdoor-poisoning/demo-mnist-poison-backdoor-adv-embedding.ipynb:1]: This can be applied to all 3 notebooks.

File: examples/tensorflow-backdoor-poisoning/src/gen_poison_model.py
Comment (jkglasbrenner) [inline examples/tensorflow-backdoor-poisoning/src/gen_poison_model.py:1]: Remove unused `download_data_archive` function from the bottom of the file.

File: examples/tensorflow-backdoor-poisoning/src/train_with_madry_pgd_defense.py
Comment (jkglasbrenner) [inline examples/tensorflow-backdoor-poisoning/src/train_with_madry_pgd_defense.py:1]: Remove unused download_data_archive function from the bottom of the file.

File: examples/tensorflow-backdoor-poisoning/src/MLproject [hunk 1]
```diff
@@ -252,23 +238,75 @@ entry_points:
...
+      --image-size {image_size}
+      --model-architecture {model_architecture}
+      --epochs {epochs}
+      --batch-size {batch_size}
+      --register-model-name {register_model_name}
+      --learning-rate {learning_rate}
+      --optimizer {optimizer}
+      --validation-split {validation_split}
+      --seed {seed}
+  train_old:
```
Comment (jkglasbrenner) [inline examples/tensorflow-backdoor-poisoning/src/MLproject:282]: If this isn't used in the notebooks, it can be removed from MLproject.

**Review by jkglasbrenner — CHANGES_REQUESTED** [review]

**Review by jkglasbrenner — APPROVED** [review]
All of the changes we discussed in my last code review have been addressed. The GitHub Actions build failure is unrelated to the changes here (the failure is due to a change in the underlying VMs hosting the public runners, they have 3GB less storage space than before, which broke the pytorch-gpu builds). Approving, and proceeding to squash and merge.
